src/api/Instance/instance.ts

```diff
@@ -37,16 +37,23 @@ instance.interceptors.request.use(
 );
 
 instance.interceptors.response.use(
-  (response) =>
-    // setToken(response.data.accessToken);
-    response,
+  (config) => config,
   async (err) => {
-    if (err.response.status === 401) {
-      const originalRequest = err.config;
-      const response = await axios.post('/auth/refresh-token');
-      localStorage.setItem('token', response.data.accessToken);
-      return instance.request(originalRequest);
+    const originalRequest = err.config;
+    if (err.response.status === 401 && originalRequest.url === `/auth/login`) {
+      return Promise.reject(err);
     }
-    return Promise.reject(err);
+    if (err.response.status === 401 && err.config && !err.config.isRetry) {
+      originalRequest.isRetry = true;
+      try {
+        const response = await instance.post('/auth/refresh-token');
+
+        saveTokenToStorage(response.data.accessToken);
+        return instance.request(originalRequest);
+      } catch (e) {
+        console.log('error', e);
```

Comment:
need to handle this case somehow. show error notification or something. Create task and assign on yourself